daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2416,6 +2417,15 @@ func (devices *DeviceSet) UnmountDevice(hash, mountPath string) error {
 	}
 	logrus.Debug("devmapper: Unmount done")
 
+	// Remove the mountpoint here. This is necessary to avoid cases where a
+	// libdm mount that is present in another namespace will cause subsequent
+	// RemoveDevice operations to fail. We ignore any errors here because this
+	// may fail on older kernels which don't have
+	// torvalds/linux@8ed936b5671bfb33d89bc60bdcc7cf0470ba52fe applied.
+	if err := system.EnsureRemoveAll(mountPath); err != nil {
+		logrus.Debugf("devmapper: error doing a removeall on unmounted device %s: %v", mountPath, err)
+	}
+
```

Comment:
Why use EnsureRemoveAll? This function is trying to do so much that I don't even understand what it is doing. After lazy unmount, why don't we simply do a `os.Remove()` ?

So idea is that use removing directory to yank the mount points which might have leaked into other mount namespaces and this will only work on newer kernels. I think this should work reasonably well. Little concerned about some use case being broken somewhere. There have been quite a few changes w.r.t referece counting. Especially make sure that the case of docker daemon being shutdown/start works while containers were running the whole time.

Reply:
Yeah, `os.Remove` is probably a better choice.

> Especially make sure that the case of docker daemon being shutdown/start works while containers were running the whole time.

Can you explain what you mean here? Are you referring to the "live restore" stuff?

Reply:
Yes, I was referring to live restore functionality.